packages/router/src/utils/config.ts

```diff
@@ -190,15 +190,33 @@ function getFullPath(parentPath: string, currentRoute: Route): string {
 }
 
 /**
- * Makes a copy of the config and adds any default required properties.
+ * The [DI token](guide/glossary/#di-token) for processing the routes when being standardized.
+ *
+ * `CUSTOM_ROUTE_PROCESSOR` is a low level API for processing routes everytime they change.
+ *
+ * @publicApi
  */
-export function standardizeConfig(r: Route): Route {
-  const children = r.children && r.children.map(standardizeConfig);
+export const CUSTOM_ROUTE_PROCESSOR = new InjectionToken<(r: Route) => Route>(
+    'CUSTOM_ROUTE_PROCESSOR',
+);
+
+
+/**
+ * Makes a copy of the config and adds any default required properties and apply the custom route
+ * processor.
+ */
+export function standardizeConfig(r: Route, customRouteProcessor?: (r: Route) => Route): Route {
+  const children = r.children && r.children.map(x => standardizeConfig(x, customRouteProcessor));
```

Comment:
Maybe, we can use Optional chaining for this one.
```suggestion
  const children = r.children?.map(x => standardizeConfig(x, customRouteProcessor));
```